### PR TITLE
Add removeObserver:fromObjectsAtIndexes:forKeyPath:context: method

### PR DESCRIPTION
**Pull request by gcasa**
Add new methods and some tests for KVO.

### CODE REVIEW

File: Source/NSKeyValueObserving.m [hunk 1]
```diff
@@ -1325,6 +1325,13 @@ - (void) removeObserver: (NSObject*)anObserver
   [iLock unlock];  
 }
 
+- (void) removeObserver: (NSObject*)anObserver
```
Comment (fredkiefer) [inline Source/NSKeyValueObserving.m:1328]: Why do we have this method here? We don't define the method without context neither and we also don't have the corresponding add methods.

Comment (gcasa) [reply Source/NSKeyValueObserving.m]: It doesn't belong there...  I discovered the method should only exist where I have it implemented now, as a category on NSArray.

**Comment by gcasa** [comment]
@fredkiefer @rfm The tests are passing now. One of the url tests seems to fail intermittently.

**Comment by gcasa** [comment]
@rfm @fredkiefer any added feedback?

**Review by rfm — APPROVED** [review]
This looks fine to me.

**Comment by gcasa** [comment]
> This looks fine to me.

Thanks. :)

**Pull request merged**